### PR TITLE
Add new class RecommendationAlgorithm and refactor algorithms

### PR DESCRIPTION
**Pull request by jilljenn**
Yeah!

### CODE REVIEW

File: mangaki/mangaki/utils/pca.py [hunk 1]
```diff
@@ -1,21 +1,15 @@
...
-from mangaki.utils.chrono import Chrono
+import numpy as np
 
 
-class MangakiPCA(object):
+class MangakiPCA(RecommendationAlgorithm):
     M = None
-    chrono = None
     def __init__(self, NB_COMPONENTS=10):
+        super().__init__
```
Comment (RaitoBezarius) [inline mangaki/mangaki/utils/pca.py]: Typo.

File: mangaki/mangaki/management/commands/compare.py [hunk 2]
```diff
@@ -12,13 +12,14 @@
...
+from mangaki.utils.pca import MangakiPCA
+from mangaki.utils.zero import MangakiZero
 
 from mangaki.utils.values import rating_values
-from mangaki.utils.zero import MangakiZero
 
 import matplotlib.pyplot as plt
 
-TEST_SIZE = 50000
+TEST_SIZE = 0.2
```
Comment (RaitoBezarius) [inline mangaki/mangaki/management/commands/compare.py]: A comment on what purpose does it serve or what does the value means? (moving from 5k to 0.2 is strange.)

File: mangaki/mangaki/utils/common.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from django.conf import settings
+from mangaki.utils.chrono import Chrono
+import pickle
+import os.path
+
+
+PICKLE_DIR = os.path.join(settings.BASE_DIR, '../pickles')
+
+
+class RecommendationAlgorithm(object):
```
Comment (RaitoBezarius) [inline mangaki/mangaki/utils/common.py]: No need to inherit from `object`. In Python 3, all classes inherits from `object` automatically.

Comment (jilljenn) [inline mangaki/mangaki/utils/common.py]: Done.

File: mangaki/mangaki/utils/common.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+from mangaki.utils.chrono import Chrono
+import pickle
+import os.path
+
+
+PICKLE_DIR = os.path.join(settings.BASE_DIR, '../pickles')
+
+
+class RecommendationAlgorithm:
+    nb_users = None
```
Comment (RaitoBezarius) [inline mangaki/mangaki/utils/common.py]: Why not put them into the `__init__ ` ?

Comment (jilljenn) [inline mangaki/mangaki/utils/common.py]: I thought it was a common way to declare a class. Isn't it?

Comment (jilljenn) [inline mangaki/mangaki/utils/common.py]: Yes. Shared by all instances.

Comment (RaitoBezarius) [inline mangaki/mangaki/utils/common.py]: AFAIK, this kind of declaration are for **class* attributes rather than instance attributes.
Is that intended ?

Comment (RaitoBezarius) [inline mangaki/mangaki/utils/common.py]: @jilljenn Alright :).

Comment (jilljenn) [inline mangaki/mangaki/utils/common.py]: I see, so maybe that's not good! Thanks.

**Pull request merged**